crates/oxc_minifier/src/peephole/minimize_conditions.rs

```diff
@@ -146,125 +133,138 @@ impl<'a> PeepholeOptimizations {
+            } else if let Some(stmt) = &mut if_stmt.alternate {
+                // "yes" is missing and "no" is not missing (and is not an expression)
+                match &mut if_stmt.test {
+                    Expression::UnaryExpression(unary_expr) if unary_expr.operator.is_not() => {
+                        if_stmt.test = ctx.ast.move_expression(&mut unary_expr.argument);
+                        if_stmt.consequent = ctx.ast.move_statement(stmt);
+                        if_stmt.alternate = None;
+                        self.mark_current_function_as_changed();
+                    }
+                    _ => {
+                        if_stmt.test = Self::minimize_not(
+                            if_stmt.test.span(),
+                            ctx.ast.move_expression(&mut if_stmt.test),
+                            ctx,
+                        );
+                        if_stmt.consequent = ctx.ast.move_statement(stmt);
+                        if_stmt.alternate = None;
+                        self.mark_current_function_as_changed();
                     }
                 }
-                Some(else_branch) => {
-                    let then_branch_is_expression_block =
-                        Self::is_foldable_express_block(then_branch);
-                    let else_branch_is_expression_block =
-                        Self::is_foldable_express_block(else_branch);
-                    // `if(foo) bar else baz` -> `foo ? bar : baz`
-                    if then_branch_is_expression_block && else_branch_is_expression_block {
-                        let test = ctx.ast.move_expression(&mut if_stmt.test);
-                        let consequent = Self::get_block_expression(&mut if_stmt.consequent, ctx);
-                        let else_branch = if_stmt.alternate.as_mut().unwrap();
-                        let alternate = Self::get_block_expression(else_branch, ctx);
-                        let expr = self.minimize_conditional(
-                            if_stmt.span,
-                            test,
-                            consequent,
-                            alternate,
+            }
+        } else {
+            // "yes" is not missing (and is not an expression)
+            if let Some(alternate) = &mut if_stmt.alternate {
+                // "yes" is not missing (and is not an expression) and "no" is not missing
+                if let Expression::UnaryExpression(unary_expr) = &mut if_stmt.test {
+                    if unary_expr.operator.is_not() {
+                        // "if (!a) return b; else return c;" => "if (a) return c; else return b;"
+                        if_stmt.test = ctx.ast.move_expression(&mut unary_expr.argument);
+                        std::mem::swap(&mut if_stmt.consequent, alternate);
+                        Self::wrap_to_avoid_ambiguous_else(if_stmt, traverse_ctx);
+                        self.mark_current_function_as_changed();
+                    }
+                }
```

Comment:
```suggestion
                }
                // "if (a) return b; else {}" => "if (a) return b;" is handled by remove_dead_code
```
As I wondered whether it is handled.